sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2531,6 +2531,8 @@ class Dataset[T] private[sql](
    */
   def inputFiles: Array[String] = {
     val files: Seq[String] = logicalPlan.collect {
+      case LogicalRelation(HadoopFsRelation(_, location: FileCatalog, _, _, _, _, _), _, _) =>
```

Comment:
Good question. It seems a call to `Dataset.inputFiles` which reads from a partitioned table will return an empty array here. On the other hand, asking for the input files for a large partitioned table will require a costly tree traversal.

I'll have to think about this some more before I have an opinion.